src/main/java/com/ldtteam/structurize/util/BlockUtils.java

```diff
@@ -553,6 +553,7 @@ else if (item instanceof BlockItem)
             // place
             world.setBlock(here, Blocks.COBBLESTONE.defaultBlockState(), Block.UPDATE_CLIENTS);
             world.setBlock(here, newState, Constants.UPDATE_FLAG);
+            world.getBlockEntity(here).applyComponentsFromItemStack(stackToPlace);
             targetBlock.setPlacedBy(world, here, newState, fakePlayer, stackToPlace);
```

Comment:
this entire branch should honestly use just the Item#useOn(...), wonder if it is even safe these days to perform blockentity-ignore blockstate update, w/e for now

Reply:
It's not so easy to even use-on without messy blockstates =D